File: python/BinarySearch.py
import math

def binary_search(arr,target):
  min = 0
  max = len(arr)
  while(min < max) :
    mid = math.floor((min+max)/2)
    midVal = arr[mid]
    if midVal == target:
      return True
    if midVal < target:
      min = mid + 1
    else:
      max = mid
  return False
    
    

File: node/BinarySearch.js
const binarySearch = (arr, target) => {
  let low = 0
  let high = arr.length
  while (low < high) {
    const mid = Math.floor(low + (high - low) / 2);
    const midVal = arr[mid]
    if (midVal === target) {
      return true
    }
    if (midVal < target) {
      low = mid + 1
    }
    if (midVal > target) {
      high = mid
    }
  }
  return false

}

module.exports = { binarySearch };